lib/privileges.ts

```diff
@@ -377,3 +378,416 @@ export const checkPrivilegesAsBoolean = (
     return false;
   }
 };
+
+const _getSubject = async (subject: { type: Extract<Subject, string>; id: string }) => {
```

Comment:
Maybe the types are messed up but looking in VScode 

Here we're saying id is a string

But looks like it could also be array of strings

<img width="800" alt="Screenshot 2024-12-09 at 10 26 25 AM" src="https://github.com/user-attachments/assets/ffdf7aec-dd19-4bfe-b309-4099a613c62a">

Also looks like sometimes the scope doesn't have a subject id it's the string "all"

is  if (subject.id === "all") a check we need?